[doc / test / mempool] unbroadcast follow-ups (#18807)
bitcoin/bitcoin: doc/release-notes.md

P2P and network changes
-----------------------

- The mempool now tracks transactions submitted via the wallet or rpcs as
  unbroadcast transactions. Every 10-15 minutes, the node will try to announce

>   unbroadcast transactions until a peer requests it via a `GETDATA`, or the

Reviewer: ```suggestion
  unbroadcast transactions until a peer requests it via a `getdata` message, or the
```

Author: updated

  transaction is removed from the mempool for other reasons. If you have a

>   custom setup to use P2P messages to submit a local transaction to your node

Reviewer: ```suggestion
  custom setup to use P2P messages to submit a transaction that belongs to a loaded wallet to your node
```

local means wallet, right?

Author: kind of.. I've been using the language "local" because it could be wallet, or submitted via `sendrawtransaction`. similarly, if we're talking about these edge cases, someone could theoretically form a transaction & give it to me out of band, then I submit from one computer to my main node via my highly specific custom P2P setup that involves being disconnected from the greater internet. but I guess every valid transaction involves some loaded wallet? 

lol. it all seems like quite an edge case. maybe here I can just remove the word "local".

new:
> If you have a custom setup to use P2P messages to submit a transaction to your node while it is not connected to other nodes, these changes reduce your initial broadcast guarantees. (#18038)

Reviewer: > If you have a custom setup to use P2P messages to submit a transaction to your node while it is not connected to other nodes, these changes reduce your initial broadcast guarantees

This has always been the case. Your pull didn't change that.

Author: oh. great point. lol ok I'll update to wallet.

Author: is there a reason you've suggested "that belongs to a loaded wallet" rather than just wallet transaction?

Author: updated to "wallet transaction", lmk if you have further thoughts

Reviewer: how would bitcoin core know that this is a wallet transaction when the wallet is not loaded? Maybe this is clear to everyone reading the sentence, but I think the distinction should be made to avoid potential confusion.

Author: ah, this was my misunderstanding. I was interpreting "loaded wallet" to mean <<wallet has spendable funds aka utxos>> vs <<node has wallet component loaded>>

also looked in the code and realized that watchonly is a wallet only thing. 

I get what you're saying now. I'm thinking about how to clarify the language.

Author: ok. recruited harding & revamped the description. @MarcoFalke what do you think now? 

@harding thanks so much for your help. I continued tinkering with the language, let me know if you have any further recommendations

  while it is not connected to other nodes, these changes reduce your initial
  broadcast guarantees. (#18038)

Updated RPCs
------------

> 

Reviewer 2: Since it seems you'll need to include the rpc info [from #18895](https://github.com/bitcoin/bitcoin/pull/18895#issuecomment-630998986) I wanted to help draft a description:
```suggestion
- `getmempoolinfo` now returns an additional `unbroadcastcount` field which represents the number of transactions that the mempool considers unbroadcast.
- Mempool RPCs such as `getmempoolentry` and `getrawmempool` with `verbose=true`now return an additional "unbroadcast" field which indicates whether the mempool considers this transaction unbroadcast.
```

Author: thanks! added with some extra info

P2P and network changes
-----------------------

> - The mempool now tracks whether transactions submitted via the wallet or rpcs

Reviewer 3: suggestion: s/rpcs/RPCs/

  have been successfully broadcast. Every 10-15 minutes, the node will try to
  announce unbroadcast transactions until a peer requests it via a `getdata`
  message or the transaction is removed from the mempool for other reasons.
  The node will not track the broadcast status of transactions submitted to the
  node using P2P relay. This version reduces the initial broadcast guarantees
  for wallet transactions submitted via P2P to a node running the wallet. (#18038)

Updated RPCs
------------

- `getmempoolinfo` now returns an additional `unbroadcastcount` field. The
  mempool tracks locally submitted transactions until their initial broadcast
  is acknowledged by a peer. This field returns the count of transactions
  waiting for acknowledgement.

- Mempool RPCs such as `getmempoolentry` and `getrawmempool` with

>   `verbose=true` now return an additional "unbroadcast" field. This indicates

Reviewer: ```suggestion
  `verbose=true` now return an additional `unbroadcast` field. This indicates
```

  whether initial broadcast of the transaction has been acknowledged by a

>   peer. `getmempoolancestors` and `getmempooldescendants` are also impacted.

Reviewer 3: I think the word "impacted" is odd here as it is often used to denote a negative connotation. Perhaps "updated", "also benefit from this change", etc.

Wallet
------

- To improve wallet privacy, the frequency of wallet rebroadcast attempts has
  been reduced from approximately once every 15 minutes to approximately once a

>   day. To maintain a similar level of guarantee for initial broadcast of wallet

Reviewer 3: suggest s/approximately once a day/once every 12-36 hours/

Author: sure. changed

Wallet
------

- To improve wallet privacy, the frequency of wallet rebroadcast attempts has

>   been reduced from approximately once every 15 minutes to once every 12-36

Reviewer 3: Each of these three release note entries employs a different verb tense: future ("will try/will not track") in the first entry, present ("now returns/tracks/returns") in the second one, and a form of past ("has been reduced") tense at the beginning of the third one. Perhaps use a consistent verb tense.

Author: hm, I updated this one to be present ("has been reduced" -> "is reduced"). 

I read through the other ones and the difference between present & future makes sense to me... eg.
> The mempool now tracks whether transactions submitted via the wallet or RPCs
  have been successfully broadcast. Every 10-15 minutes, the node will try to
  announce unbroadcast transactions until...

the present "now tracks" indicates how the code has changed, but the future "will try" is what happens when you run the code in the future. 

so I've left them for now. if you have any specific suggestions, I'm happy to consider them.
